opm/simulators/flow/DamarisWriter.hpp

```diff
@@ -149,25 +152,56 @@ class DamarisWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::G
                    simulator.vanguard().eclState(),
                    simulator.vanguard().summaryConfig(),
                    simulator.vanguard().grid(),
-                   simulator.vanguard().grid().comm().rank() == 0 ? &simulator.vanguard().equilGrid() : nullptr,
+                   ((simulator.vanguard().grid().comm().rank() == 0)
+                    ? &simulator.vanguard().equilGrid()
+                    : nullptr),
                    simulator.vanguard().gridView(),
                    simulator.vanguard().cartesianIndexMapper(),
-                   simulator.vanguard().grid().comm().rank() == 0 ? &simulator.vanguard().equilCartesianIndexMapper() : nullptr,
+                   ((simulator.vanguard().grid().comm().rank() == 0)
+                    ? &simulator.vanguard().equilCartesianIndexMapper()
+                    : nullptr),
                    false, false)
         , simulator_(simulator)
     {
         this->damarisUpdate_ = true ;
 
-        rank_ = simulator_.vanguard().grid().comm().rank() ;
-        nranks_ = simulator_.vanguard().grid().comm().size();
+        this->rank_ = this->simulator_.vanguard().grid().comm().rank() ;
+        this->nranks_ = this->simulator_.vanguard().grid().comm().size();
+
+        this->elements_rank_offsets_.resize(this->nranks_);
         
-        const auto& gridView = simulator_.gridView();
-        const auto& interior_elements = elements(gridView, Dune::Partitions::interior);
         // Get the size of the unique vector elements (excludes the shared 'ghost' elements)
-        numElements_ = std::distance(interior_elements.begin(), interior_elements.end());
+        //
+        // Might possibly use
+        //
+        //     detail::countLocalInteriorCellsGridView(this->simulator_.gridView())
+        //
+        // from countGlobalCells.hpp instead of calling std::distance() directly.
+        {
+            const auto& gridView = this->simulator_.gridView();
+            const auto& interior_elements = elements(gridView, Dune::Partitions::interior);
+
+            this->numElements_ = std::distance(interior_elements.begin(), interior_elements.end());
+        }
 
-        this->elements_rank_offsets_.resize(nranks_) ;
-        this->damarisOutputModule_ = std::make_unique<OutputBlackOilModule<TypeTag>>(simulator, this->collectOnIORank_);
+        if (this->nranks_ > 1) {
```

Comment:
consider a helper in ParallelSerialization ala eclStateBroadcast.

Reply:
> consider helper in ParallelSerialization (eclBroadCast).

Good idea.  I've pushed an update to that effect.